GSON parse error when fetching accounts by trustline

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version are you using?
0.31.0

### What did you do?
Fetch accounts based on trustline. This error happens with any asset/issuer combo. Can try with any live one, such as:
AAVE
GCQEKCSJ7IQTEEOT57T5LXKEYPPEBZGODQZHUYOBW6HKNSXUXAMH376F


`        AssetTypeCreditAlphaNum asset =
                (assetCode.length() < 5)
                ? new AssetTypeCreditAlphaNum4(assetCode, assetIssuer)
                : new AssetTypeCreditAlphaNum12(assetCode, assetIssuer);
        Server server = new Server("https://horizon.stellar.org");
        Page<AccountResponse> accountsPages = server.accounts()
                                                       .forAsset(asset)
                                                       .limit(20)
                                                       .execute();
`

### What did you expect to see?
Successful fetch of data

### What did you see instead?
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is shadow.com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.records[35].balances[0].liquidity_pool_id

This code worked fine in 0.28.0, it also appears to be working fine when running 0.31.0 on TESTNET, but it does not work when running on live system.

Full stack trace:
`
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is shadow.com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.records[35].balances[0].liquidity_pool_id
                at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014)
                at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:497)
                at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:584)
                at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
                at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
                at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
                at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
                at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
                at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
                at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
                at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
                at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
                at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
                at org.springframework.cloud.sleuth.instrument.web.servlet.TracingFilter.doFilter(TracingFilter.java:89)
                at org.springframework.cloud.sleuth.autoconfig.instrument.web.LazyTracingFilter.doFilter(TraceWebServletConfiguration.java:114)
                at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
                at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
                at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
                at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
                at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
                at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
                at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
                at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
                at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
                at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
                at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
                at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:117)
                at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
                at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
                at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
                at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
                at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
                at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
                at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
                at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
                at io.undertow.servlet.handlers.SendErrorPageHandler.handleRequest(SendErrorPageHandler.java:52)
                at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
                at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:269)
                at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:78)
                at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:133)
                at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:130)
                at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
                at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
                at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:249)
                at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:78)
                at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:99)
                at io.undertow.server.Connectors.executeRootHandler(Connectors.java:387)
                at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:841)
                at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
                at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2019)
                at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1558)
                at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1423)
                at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: shadow.com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.records[35].balances[0].liquidity_pool_id
                at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226)
                at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
                at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
                at shadow.com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
                at shadow.com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72)
                at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
                at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
                at shadow.com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
                at shadow.com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
                at shadow.com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
                at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
                at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
                at shadow.com.google.gson.Gson.fromJson(Gson.java:927)
                at shadow.com.google.gson.Gson.fromJson(Gson.java:994)
                at org.stellar.sdk.responses.PageDeserializer.deserialize(PageDeserializer.java:50)
                at org.stellar.sdk.responses.PageDeserializer.deserialize(PageDeserializer.java:19)
                at shadow.com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
                at shadow.com.google.gson.Gson.fromJson(Gson.java:927)
                at shadow.com.google.gson.Gson.fromJson(Gson.java:892)
                at shadow.com.google.gson.Gson.fromJson(Gson.java:841)
                at org.stellar.sdk.requests.ResponseHandler.handleResponse(ResponseHandler.java:42)
                at org.stellar.sdk.requests.AccountsRequestBuilder.execute(AccountsRequestBuilder.java:163)
                at org.stellar.sdk.requests.AccountsRequestBuilder.execute(AccountsRequestBuilder.java:188)
…
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
                at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.base/java.lang.reflect.Method.invoke(Method.java:568)
                at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:197)
                at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:141)
                at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:106)
                at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:894)
                at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
                at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
                at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1063)
                at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)
                at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
                ... 54 more
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.records[35].balances[0].liquidity_pool_id
                at shadow.com.google.gson.internal.bind.JsonTreeReader.expect(JsonTreeReader.java:162)
                at shadow.com.google.gson.internal.bind.JsonTreeReader.beginObject(JsonTreeReader.java:87)
                at shadow.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215)
                ... 90 more
`

--- a/src/main/java/org/stellar/sdk/responses/PageDeserializer.java
+++ b/src/main/java/org/stellar/sdk/responses/PageDeserializer.java
@@ -10,6 +10,7 @@ import com.google.gson.JsonParseException;
 import com.google.gson.reflect.TypeToken;
 
 import org.stellar.sdk.Asset;
+import org.stellar.sdk.LiquidityPoolID;
 import org.stellar.sdk.Predicate;
 import org.stellar.sdk.responses.effects.EffectResponse;
 import org.stellar.sdk.responses.operations.OperationResponse;
@@ -45,6 +46,7 @@ class PageDeserializer<E> implements JsonDeserializer<Page<E>> {
             .registerTypeAdapter(EffectResponse.class, new EffectDeserializer())
             .registerTypeAdapter(LiquidityPoolResponse.class, new LiquidityPoolDeserializer())
             .registerTypeAdapter(TransactionResponse.class, new TransactionDeserializer())
+            .registerTypeAdapter(LiquidityPoolID.class, new LiquidityPoolIDDeserializer())
             .create();
 
     return gson.fromJson(newJson, pageType.getType());

--- a/src/test/java/org/stellar/sdk/responses/AccountsPageDeserializerTest.java
+++ b/src/test/java/org/stellar/sdk/responses/AccountsPageDeserializerTest.java
@@ -1,10 +1,12 @@
 package org.stellar.sdk.responses;
 
+import org.junit.Test;
+import org.stellar.sdk.LiquidityPoolID;
+
+import com.google.common.base.Optional;
 import com.google.gson.reflect.TypeToken;
 
 import junit.framework.TestCase;
-
-import org.junit.Test;
 
 public class AccountsPageDeserializerTest extends TestCase {
   @Test
@@ -16,6 +18,16 @@ public class AccountsPageDeserializerTest extends TestCase {
     assertEquals(accountsPage.getLinks().getNext().getHref(), "/accounts?order=asc&limit=10&cursor=86861418598401");
     assertEquals(accountsPage.getLinks().getPrev().getHref(), "/accounts?order=desc&limit=10&cursor=1");
     assertEquals(accountsPage.getLinks().getSelf().getHref(), "/accounts?order=asc&limit=10&cursor=");
+  }
+  
+  @Test
+  public void testDeserializeWithLiquidityPoolBalance() {
+    Page<AccountResponse> accountsPage = GsonSingleton.getInstance().fromJson(jsonLiquidityPool, new TypeToken<Page<AccountResponse>>() {}.getType());
+
+    assertEquals(accountsPage.getRecords().get(0).getAccountId(), "GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST");
+    
+        
+    assertEquals(accountsPage.getRecords().get(0).getBalances()[0].getLiquidityPoolID(), Optional.of(new LiquidityPoolID("a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088")));
   }
 
   String json = "{\n" +
@@ -85,4 +97,116 @@ public class AccountsPageDeserializerTest extends TestCase {
         "    }\n" +
         "  }\n" +
         "}";
+  
+  String jsonLiquidityPool = "{\n" + 
+          "  \"_links\": {\n" + 
+          "    \"self\": {\n" + 
+          "      \"href\": \"https://horizon.publicnode.org/accounts/?asset=USDC%3AGA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN&cursor=&limit=1&order=desc\"\n" + 
+          "    },\n" + 
+          "    \"next\": {\n" + 
+          "      \"href\": \"https://horizon.publicnode.org/accounts/?asset=USDC%3AGA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN&cursor=GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST&limit=1&order=desc\"\n" + 
+          "    },\n" + 
+          "    \"prev\": {\n" + 
+          "      \"href\": \"https://horizon.publicnode.org/accounts/?asset=USDC%3AGA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN&cursor=GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST&limit=1&order=asc\"\n" + 
+          "    }\n" + 
+          "  },\n" + 
+          "  \"_embedded\": {\n" + 
+          "    \"records\": [\n" + 
+          "      {\n" + 
+          "        \"_links\": {\n" + 
+          "          \"self\": {\n" + 
+          "            \"href\": \"https://horizon.publicnode.org/accounts/GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST\"\n" + 
+          "          },\n" + 
+          "          \"transactions\": {\n" + 
+          "            \"href\": \"https://horizon.publicnode.org/accounts/GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST/transactions{?cursor,limit,order}\",\n" + 
+          "            \"templated\": true\n" + 
+          "          },\n" + 
+          "          \"operations\": {\n" + 
+          "            \"href\": \"https://horizon.publicnode.org/accounts/GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST/operations{?cursor,limit,order}\",\n" + 
+          "            \"templated\": true\n" + 
+          "          },\n" + 
+          "          \"payments\": {\n" + 
+          "            \"href\": \"https://horizon.publicnode.org/accounts/GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST/payments{?cursor,limit,order}\",\n" + 
+          "            \"templated\": true\n" + 
+          "          },\n" + 
+          "          \"effects\": {\n" + 
+          "            \"href\": \"https://horizon.publicnode.org/accounts/GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST/effects{?cursor,limit,order}\",\n" + 
+          "            \"templated\": true\n" + 
+          "          },\n" + 
+          "          \"offers\": {\n" + 
+          "            \"href\": \"https://horizon.publicnode.org/accounts/GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST/offers{?cursor,limit,order}\",\n" + 
+          "            \"templated\": true\n" + 
+          "          },\n" + 
+          "          \"trades\": {\n" + 
+          "            \"href\": \"https://horizon.publicnode.org/accounts/GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST/trades{?cursor,limit,order}\",\n" + 
+          "            \"templated\": true\n" + 
+          "          },\n" + 
+          "          \"data\": {\n" + 
+          "            \"href\": \"https://horizon.publicnode.org/accounts/GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST/data/{key}\",\n" + 
+          "            \"templated\": true\n" + 
+          "          }\n" + 
+          "        },\n" + 
+          "        \"id\": \"GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST\",\n" + 
+          "        \"account_id\": \"GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST\",\n" + 
+          "        \"sequence\": \"166506652181200896\",\n" + 
+          "        \"subentry_count\": 5,\n" + 
+          "        \"last_modified_ledger\": 38767851,\n" + 
+          "        \"last_modified_time\": \"2021-12-17T20:22:00Z\",\n" + 
+          "        \"thresholds\": {\n" + 
+          "          \"low_threshold\": 20,\n" + 
+          "          \"med_threshold\": 20,\n" + 
+          "          \"high_threshold\": 20\n" + 
+          "        },\n" + 
+          "        \"flags\": {\n" + 
+          "          \"auth_required\": false,\n" + 
+          "          \"auth_revocable\": false,\n" + 
+          "          \"auth_immutable\": false,\n" + 
+          "          \"auth_clawback_enabled\": false\n" + 
+          "        },\n" + 
+          "        \"balances\": [\n" + 
+          "        {\n" + 
+          "            \"balance\": \"974462.6017340\",\n" + 
+          "            \"liquidity_pool_id\": \"a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088\",\n" + 
+          "            \"limit\": \"922337203685.4775807\",\n" + 
+          "            \"last_modified_ledger\": 38809512,\n" + 
+          "            \"is_authorized\": false,\n" + 
+          "            \"is_authorized_to_maintain_liabilities\": false,\n" + 
+          "            \"asset_type\": \"liquidity_pool_shares\"\n" + 
+          "            },\n" + 
+          "          {\n" + 
+          "            \"balance\": \"0.0000000\",\n" + 
+          "            \"limit\": \"922337203685.4775807\",\n" + 
+          "            \"buying_liabilities\": \"0.0000000\",\n" + 
+          "            \"selling_liabilities\": \"0.0000000\",\n" + 
+          "            \"sponsor\": \"GCZGSFPITKVJPJERJIVLCQK5YIHYTDXCY45ZHU3IRPBP53SXSCALTEST\",\n" + 
+          "            \"last_modified_ledger\": 38767851,\n" + 
+          "            \"is_authorized\": true,\n" + 
+          "            \"is_authorized_to_maintain_liabilities\": true,\n" + 
+          "            \"asset_type\": \"credit_alphanum4\",\n" + 
+          "            \"asset_code\": \"USDC\",\n" + 
+          "            \"asset_issuer\": \"GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN\"\n" + 
+          "          },\n" + 
+          "          {\n" + 
+          "            \"balance\": \"0.0000000\",\n" + 
+          "            \"buying_liabilities\": \"0.0000000\",\n" + 
+          "            \"selling_liabilities\": \"0.0000000\",\n" + 
+          "            \"asset_type\": \"native\"\n" + 
+          "          }\n" + 
+          "        ],\n" + 
+          "        \"signers\": [\n" + 
+          "          {\n" + 
+          "            \"weight\": 10,\n" + 
+          "            \"key\": \"GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST\",\n" + 
+          "            \"type\": \"ed25519_public_key\"\n" + 
+          "          }\n" + 
+          "        ],\n" + 
+          "        \"data\": {},\n" + 
+          "        \"num_sponsoring\": 0,\n" + 
+          "        \"num_sponsored\": 7,\n" + 
+          "        \"sponsor\": \"GCZGSFPITKVJPJERJIVLCQK5YIHYTDXCY45ZHU3IRPBP53SXSCALTEST\",\n" + 
+          "        \"paging_token\": \"GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST\"\n" + 
+          "      }\n" + 
+          "    ]\n" + 
+          "  }\n" + 
+          "}";
 }

--- a/src/test/java/org/stellar/sdk/responses/AccountsPageDeserializerTest.java
+++ b/src/test/java/org/stellar/sdk/responses/AccountsPageDeserializerTest.java
@@ -25,9 +25,8 @@ public class AccountsPageDeserializerTest extends TestCase {
     Page<AccountResponse> accountsPage = GsonSingleton.getInstance().fromJson(jsonLiquidityPool, new TypeToken<Page<AccountResponse>>() {}.getType());
 
     assertEquals(accountsPage.getRecords().get(0).getAccountId(), "GDZZYLIIJ24HWAVWAQ2PJVNKHUJLJVVLFY2SSLYINBHDY5KZTLPTEST");
-    
-        
     assertEquals(accountsPage.getRecords().get(0).getBalances()[0].getLiquidityPoolID(), Optional.of(new LiquidityPoolID("a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088")));
+    assertEquals(accountsPage.getRecords().get(0).getBalances()[1].getLiquidityPoolID(), Optional.absent());
   }
 
   String json = "{\n" +

--- a/src/test/java/org/stellar/sdk/responses/TradesPageDeserializerTest.java
+++ b/src/test/java/org/stellar/sdk/responses/TradesPageDeserializerTest.java
@@ -5,6 +5,7 @@ import com.google.gson.reflect.TypeToken;
 import junit.framework.TestCase;
 import org.junit.Test;
 import org.stellar.sdk.AssetTypeNative;
+import org.stellar.sdk.LiquidityPoolID;
 
 import static java.lang.Long.valueOf;
 import static org.stellar.sdk.Asset.create;
@@ -28,8 +29,20 @@ public class TradesPageDeserializerTest extends TestCase {
         assertEquals(tradesPage.getRecords().get(0).getCounterAsset(), create(null,"JPY", "GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM"));
         assertEquals(tradesPage.getRecords().get(0).getPrice().getNumerator(), valueOf(267));
         assertEquals(tradesPage.getRecords().get(0).getPrice().getDenominator(), valueOf(1000));
+        assertEquals(tradesPage.getRecords().get(0).getCounterLiquidityPoolID(), Optional.absent());
+        assertEquals(tradesPage.getRecords().get(0).getBaseLiquidityPoolID(), Optional.absent());
 
         assertEquals(tradesPage.getRecords().get(1).getBaseAccount(), Optional.of("GAVH5JM5OKXGMQDS7YPRJ4MQCPXJUGH26LYQPQJ4SOMOJ4SXY472ZM7G"));
+    }
+    @Test
+    public void testDeserializeLiquidityPool() {
+        Page<TradeResponse> tradesPage = GsonSingleton.getInstance().fromJson(jsonLiquidityPool, new TypeToken<Page<TradeResponse>>() {}.getType());
+        
+        
+        assertEquals(tradesPage.getRecords().get(0).getBaseLiquidityPoolID(), Optional.of(new LiquidityPoolID("a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7bbase")));
+        assertEquals(tradesPage.getRecords().get(0).getCounterLiquidityPoolID(), Optional.absent());
+        assertEquals(tradesPage.getRecords().get(1).getBaseLiquidityPoolID(), Optional.absent());
+        assertEquals(tradesPage.getRecords().get(1).getCounterLiquidityPoolID(), Optional.of(new LiquidityPoolID("a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088")));
     }
 
     String json = "{\n" +
@@ -381,4 +394,97 @@ public class TradesPageDeserializerTest extends TestCase {
             "    ]\n" +
             "  }\n" +
             "}";
+    
+    
+    String jsonLiquidityPool = 
+            "{\n" + 
+            "  \"_links\": {\n" + 
+            "    \"self\": {\n" + 
+            "      \"href\": \"https://horizon.stellar.org/liquidity_pools/a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088/trades?cursor=&limit=10&order=asc\"\n" + 
+            "    },\n" + 
+            "    \"next\": {\n" + 
+            "      \"href\": \"https://horizon.stellar.org/liquidity_pools/a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088/trades?cursor=163706513893126145-0&limit=10&order=asc\"\n" + 
+            "    },\n" + 
+            "    \"prev\": {\n" + 
+            "      \"href\": \"https://horizon.stellar.org/liquidity_pools/a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088/trades?cursor=163706170297630721-3&limit=10&order=desc\"\n" + 
+            "    }\n" + 
+            "  },\n" + 
+            "  \"_embedded\": {\n" + 
+            "    \"records\": [\n" + 
+            "      {\n" + 
+            "        \"_links\": {\n" + 
+            "          \"self\": {\n" + 
+            "            \"href\": \"\"\n" + 
+            "          },\n" + 
+            "          \"base\": {\n" + 
+            "            \"href\": \"https://horizon.stellar.org/liquidity_pools/a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088\"\n" + 
+            "          },\n" + 
+            "          \"counter\": {\n" + 
+            "            \"href\": \"https://horizon.stellar.org/accounts/GAVYIUWQFQX2GJ7FWNGVXSSLMJO75V5XY5M2YON4DKYD24LPB6QLCERP\"\n" + 
+            "          },\n" + 
+            "          \"operation\": {\n" + 
+            "            \"href\": \"https://horizon.stellar.org/operations/163706170297630721\"\n" + 
+            "          }\n" + 
+            "        },\n" + 
+            "        \"id\": \"163706170297630721-3\",\n" + 
+            "        \"paging_token\": \"163706170297630721-3\",\n" + 
+            "        \"ledger_close_time\": \"2021-11-03T15:00:52Z\",\n" + 
+            "        \"trade_type\": \"liquidity_pool\",\n" + 
+            "        \"liquidity_pool_fee_bp\": 30,\n" + 
+            "        \"base_liquidity_pool_id\": \"a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7bbase\",\n" + 
+            "        \"base_amount\": \"0.0010000\",\n" + 
+            "        \"base_asset_type\": \"native\",\n" + 
+            "        \"counter_offer_id\": \"4775392188725018625\",\n" + 
+            "        \"counter_account\": \"GAVH5JM5OKXGMQDS7YPRJ4MQCPXJUGH26LYQPQJ4SOMOJ4SXY472ZM7G\",\n" + 
+            "        \"counter_amount\": \"0.0003679\",\n" + 
+            "        \"counter_asset_type\": \"credit_alphanum4\",\n" + 
+            "        \"counter_asset_code\": \"USDC\",\n" + 
+            "        \"counter_asset_issuer\": \"GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN\",\n" + 
+            "        \"base_is_seller\": true,\n" + 
+            "        \"price\": {\n" + 
+            "          \"n\": \"3679\",\n" + 
+            "          \"d\": \"10000\"\n" + 
+            "        }\n" + 
+            "      },\n" + 
+            
+            "      {\n" + 
+            "        \"_links\": {\n" + 
+            "          \"self\": {\n" + 
+            "            \"href\": \"\"\n" + 
+            "          },\n" + 
+            "          \"base\": {\n" + 
+            "            \"href\": \"https://horizon.stellar.org/accounts/GA7BRV2K3OM27NLVY2IJQGYZEQ7AEZSPR3Y3XA6COHPMOPOEUHQBDYBC\"\n" + 
+            "          },\n" + 
+            "          \"counter\": {\n" + 
+            "            \"href\": \"https://horizon.stellar.org/liquidity_pools/a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088\"\n" + 
+            "          },\n" + 
+            "          \"operation\": {\n" + 
+            "            \"href\": \"https://horizon.stellar.org/operations/163706436584558593\"\n" + 
+            "          }\n" + 
+            "        },\n" + 
+            "        \"id\": \"163706436584558593-1\",\n" + 
+            "        \"paging_token\": \"163706436584558593-1\",\n" + 
+            "        \"ledger_close_time\": \"2021-11-03T15:07:00Z\",\n" + 
+            "        \"trade_type\": \"liquidity_pool\",\n" + 
+            "        \"liquidity_pool_fee_bp\": 30,\n" + 
+            "        \"base_offer_id\": \"4775392455011946497\",\n" + 
+            "        \"base_account\": \"GAVH5JM5OKXGMQDS7YPRJ4MQCPXJUGH26LYQPQJ4SOMOJ4SXY472ZM7G\",\n" + 
+            "        \"base_amount\": \"0.0002207\",\n" + 
+            "        \"base_asset_type\": \"native\",\n" + 
+            "        \"counter_liquidity_pool_id\": \"a468d41d8e9b8f3c7209651608b74b7db7ac9952dcae0cdf24871d1d9c7b0088\",\n" + 
+            "        \"counter_amount\": \"0.0000831\",\n" + 
+            "        \"counter_asset_type\": \"credit_alphanum4\",\n" + 
+            "        \"counter_asset_code\": \"USDC\",\n" + 
+            "        \"counter_asset_issuer\": \"GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN\",\n" + 
+            "        \"base_is_seller\": false,\n" + 
+            "        \"price\": {\n" + 
+            "          \"n\": \"831\",\n" + 
+            "          \"d\": \"2207\"\n" + 
+            "        }\n" + 
+            "      },\n" + 
+            "    ]\n" + 
+            "  }\n" + 
+            "}\n" + 
+            "\n";
+
 }

